pkg/envoy/server.go

```diff
@@ -786,36 +786,18 @@ func (s *XDSServer) UpdateNetworkPolicy(ep logger.EndpointUpdater, policy *polic
 		policies = append(policies, networkPolicy)
 	}
 
-	if ep.HasSidecarProxy() { // Use sidecar proxy.
-		// If there are no L7 rules, we expect Envoy to NOT be configured with
-		// an L7 filter, in which case we'd never receive an ACK for the policy,
-		// and we'd wait forever.
-		// TODO: Remove this when we implement and inject an Envoy network filter
-		// into every Envoy listener to filter at L3/L4.
-		var hasL7Rules bool
-	Policies:
```

Comment:
Can you elaborate on why it is safe to remove this check whose purpose is to set the waitgroup to nil if we are running with a sidecar proxy?